### PR TITLE
dconf: minor refactor

### PR DESCRIPTION
**Pull request by russoz**
##### SUMMARY

Minor refactor to use `module_utils.deps` and `required_if`.

##### ISSUE TYPE

- Refactoring Pull Request

##### COMPONENT NAME
dconf

### CODE REVIEW

File: plugins/modules/dconf.py [hunk 1]
```diff
@@ -414,7 +409,10 @@ def main():
             # Converted to str below after special handling of bool.
             value=dict(required=False, default=None, type='raw'),
         ),
-        supports_check_mode=True
+        supports_check_mode=True,
+        required_if=[
+            ('state', 'present', ['value'])
```
Comment (felixfontein) [inline plugins/modules/dconf.py:414]:
```suggestion
            ('state', 'present', ['value']),
```

**Review by felixfontein — APPROVED** [review]

**Comment by felixfontein** [comment]
@russoz thanks!

**Comment by pboushy** [comment]
Hey @russoz - this PR appears to have added dependencies to `dconf` on other files within community without adding significant benefit.

Why was it decided to add this additional level of dependency?

**Comment by felixfontein** [comment]
Replacing
```python
PSUTIL_IMP_ERR = None
try:
    import psutil
    HAS_PSUTIL = True
except ImportError:
    PSUTIL_IMP_ERR = traceback.format_exc()
    HAS_PSUTIL = False

[...]

    if not HAS_PSUTIL:
        module.fail_json(msg=missing_required_lib("psutil"), exception=PSUTIL_IMP_ERR)
```
(similar code which exists all over the collection in various degrees of similarity) with
```python
with deps.declare("psutil"):
    import psutil

[...]

    deps.validate(module)
```
(which allows to better standarize missing library handling without having to repeat too much code everywhere) looks like a significant benefit to me.

**Pull request merged**